app/controllers/bonus_engine/api/points_controller.rb

```diff
@@ -1,34 +1,59 @@
 module BonusEngine
   module Api
     class PointsController < BaseController
-      def create
-        point = BonusEngine::Point.new create_point_params
+      before_filter :check_balance, only: [:create]
+      before_filter :check_balance_for_update, only: [:update]
 
-        if point.save
-          render nothing: true, status: :created
+      def create
+        @point = BonusEngine::Point.new create_point_params
+        if @point.save
+          stats = @event.stats_for current_user
+          @balance = stats[:balance]
+          @pending = stats[:pending]
+          render :create, status: :created
         else
-          render json: point.errors, status: :unprocessable_entity
+          render json: @point.errors, status: :unprocessable_entity
         end
       end
 
       def update
-        point = BonusEngine::Point.find params[:id]
+        @point = BonusEngine::Point.find params[:id]
 
-        if point.update(create_point_params)
-          render nothing: true, status: :ok
+        if @point.update(create_point_params)
+          stats = @event.stats_for current_user
```

Comment:
You could dry this ... if you take a look, the actions you do on your update and your save are pretty much the same; at least, I think you could move these lines:
```
stats = @event.stats_for current_user
@balance = stats[:balance]
@pending = stats[:pending]
```
to another private method, something like:
```
def set_stats
  stats = @event.stats_for current_user
  @balance, @pending = stats.values_at(:balance, :pending)
end
```

so both of your actions would become:
```
def create 
   ...
    if @point.save ...
        set_stats
        render :create .
    ....
   ...
end
```